client/state/inline-help/test/reducer.js

```diff
@@ -32,4 +71,335 @@ describe( 'reducer', () => {
 			} );
 		} );
 	} );
+
+	describe( '#ui()', () => {
+		test( 'should correct set isVisible prop to true', () => {
+			const state = ui( null, {
+				type: INLINE_HELP_SHOW,
+			} );
+
+			expect( state ).to.eql( {
+				isVisible: true,
+			} );
+		} );
+
+		test( 'should correct set isVisible prop to false', () => {
+			const original = deepFreeze( { isVisible: true } );
+			const state = ui( original, {
+				type: INLINE_HELP_HIDE,
+			} );
+
+			expect( state ).to.eql( {
+				isVisible: false,
+			} );
+		} );
+	} );
+
+	describe( '#requesting()', () => {
+		test( 'should correctly set boolean flag to true for query key', () => {
+			const testQuery = 'testQueryKey';
+			const state = requesting( null, {
+				type: INLINE_HELP_SEARCH_REQUEST,
+				searchQuery: testQuery,
+			} );
+
+			expect( state ).to.eql( {
+				[ testQuery ]: true,
+			} );
+		} );
+
+		test( 'should correctly set boolean flag to false for query key on success', () => {
+			const testQuery = 'testQueryKey';
+			const original = deepFreeze( { [ testQuery ]: true } );
+			const state = requesting( original, {
+				type: INLINE_HELP_SEARCH_REQUEST_SUCCESS,
+				searchQuery: testQuery,
+			} );
+
+			expect( state ).to.eql( {
+				[ testQuery ]: false,
+			} );
+		} );
+
+		test( 'should correctly set boolean flag to false for query key on failure', () => {
+			const testQuery = 'testQueryKey';
+			const original = deepFreeze( { [ testQuery ]: true } );
+			const state = requesting( original, {
+				type: INLINE_HELP_SEARCH_REQUEST_FAILURE,
+				searchQuery: testQuery,
+			} );
+
+			expect( state ).to.eql( {
+				[ testQuery ]: false,
+			} );
+		} );
+	} );
+
+	describe( '#search()', () => {
+		test.each( [ 'testQueryKey', '' ] )(
+			'should correctly set the current search query text',
+			( testQuery ) => {
+				const state = search( null, {
+					type: INLINE_HELP_SEARCH_REQUEST,
+					searchQuery: testQuery,
+				} );
+
+				expect( state ).to.eql( {
+					searchQuery: testQuery,
+				} );
+			}
+		);
+
+		test( 'should correctly set search results keyed by search term', () => {
+			const firstQuery = 'testQuery';
+			const secondQuery = 'anotherQuery';
+
+			let state = search(
+				{},
+				{
+					type: INLINE_HELP_SEARCH_REQUEST_SUCCESS,
+					searchQuery: firstQuery,
+					searchResults: API_RESULT_FIXTURE,
```

Comment:
should it be deep-frozen?

Reply:
Actually this is a great point. Several tests are using initial state or updating state without protection of `deepFreeze` to assert that state is not being mutated in place.

Now added to all tests to assert against mutations.